File: src/Core.Domain/Entities/MetodoPago.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TurnosApp.Core.Domain.Entities;

using TurnosApp.Core.Domain.Common;
using TurnosApp.Core.Domain.Enums;

public class MetodoPago : TenantEntity
{
    public string Nombre { get; set; } = string.Empty;
    public TipoModificadorPago TipoModificador { get; set; } = TipoModificadorPago.Ninguno;

    // Siempre positivo — el signo del efecto sobre el cliente lo da TipoModificador.
    public decimal PorcentajeModificador { get; set; }

    // Siempre positivo — costo real que le cobra el proveedor de pago al negocio.
    // No afecta lo que paga el cliente, solo la ganancia neta.
    public decimal PorcentajeComision { get; set; }

    public bool Activo { get; set; } = true;

    // Distingue lo único que se cuenta físicamente en un arqueo de caja de lo que solo se concilia después.
    public bool EsEfectivo { get; set; }

    // Navegación
    public ICollection<Cobro> Cobros { get; set; } = [];
}

File: turnos-app-frontend/src/types/MetodoPago.ts
// src/types/MetodoPago.ts

// Coincide con el enum TipoModificadorPago del backend (src/Core.Domain/Enums/TipoModificadorPago.cs)
export type TipoModificadorPago = 'Ninguno' | 'Bonificacion' | 'Recargo';

export interface MetodoPago {
  id: number;
  nombre: string;
  tipoModificador: TipoModificadorPago;
  porcentajeModificador: number;
  porcentajeComision: number;
  activo: boolean;
  esEfectivo: boolean;
}

export interface MetodoPagoCreateDTO {
  nombre: string;
  tipoModificador: TipoModificadorPago;
  porcentajeModificador: number;
  porcentajeComision: number;
  esEfectivo: boolean;
}

export interface MetodoPagoUpdateDTO extends MetodoPagoCreateDTO {
  activo: boolean;
}
